examples/advanced/torque-controls.html

```diff
@@ -0,0 +1,188 @@
+            container: 'map',
+            style: {
+                version: 8,
+                sources: {},
+                layers: [{
+                    id: 'background',
+                    type: 'background',
+                    paint: {
+                        'background-color': 'black'
+                    }
+                }]
+            },
+            center: [0, 0],
+            zoom: 0,
+            dragRotate: false
+        });
+
+        const animStart = '2018-04-11T12:00:00Z';
+        const animEnd = '2018-04-12T12:00:00Z';
+
+        const source = new carto.source.GeoJSON(generateData(), {
+            dateColumns: ['sim_time']
+        });
+
+        function generateData() {
+            const features = [];
+            const length = 200;
+            const n = 50;
+            const x0 = 0;
+            const y0 = 0;
+            let id = 1;
+            const min_st = new Date(animStart);
+            const max_st = new Date(animEnd);
+            for (let step = 1; step <= n; ++step) {
+                features.push({
+                    type: 'Feature',
+                    geometry: {
+                        type: 'Point',
+                        coordinates: [step / n * length + x0 - length / 2, y0]
+                    },
+                    properties: {
+                        cartodb_id: id++,
+                        sim_time: (min_st.getTime() + (max_st - min_st) * step / n)
+                    }
+                });
+            }
+            return {
+                type: 'FeatureCollection',
+                features
+            };
+        }
+
+
+        const viz = new carto.Viz(`
+            @duration: 10
+            @torque: torque($sim_time, @duration, fade(0.2, 0.2))
+            width: linear($sim_time) * 100 + 1
+            strokeWidth: 0
+            filter: @torque`
+        );
+
+        const layer = new carto.Layer('layer', source, viz);
+        layer.addTo(map, 'background');
+
+        // Elements
+        const $playBtn = document.getElementById('js-play-btn');
+        const $stopBtn = document.getElementById('js-stop-btn');
+        const $pauseBtn = document.getElementById('js-pause-btn');
+        const $durationRange = document.getElementById('js-duration-range');
+        const $timeRange = document.getElementById('js-time-range');
+        const $spanDuration = document.getElementById('js-duration-span');
+
+        // Callbacks
+        $durationRange.onchange = () => {
+            const duration = parseInt($durationRange.value, 10);
+            viz.variables.duration = $spanDuration.innerHTML = duration;
+        }
+
+        $timeRange.onchange = () => {
+            viz.variables.torque.setSimProgress($timeRange.value);
+        }
+
+        $playBtn.onclick = () => viz.variables.torque.play();
+        $pauseBtn.onclick = () => viz.variables.torque.pause();
+        $stopBtn.onclick = () => viz.variables.torque.stop();
+
+        setInterval(() => {
```

Comment:
Can't we use `layer.on(''updated',...)`?

Reply:
nope, the frequency is so damn high so the range-slider events are overridden

Reply:
You can use

Reply:
You can use:

Reply:
You can use:
```
        let last = $timeRange.value;
        $timeRange.onchange = (ev) => {
            viz.variables.torque.setSimProgress($timeRange.value);
            last = $timeRange.value;
        }
        layer.on('updated', () => {
            if ($timeRange.value == last) {
                $timeRange.value = viz.variables.torque.getSimProgress();
                last = $timeRange.value;
            }
        });
```